### PR TITLE
[zh] Move kubectl overview to be section index

### PR DESCRIPTION
**Pull request by Sea-n**
Since #29847, `/docs/reference/kubectl/overview` is migrated to `/docs/reference/kubectl`.
This PR also added link anchors, resync with en content and fix some formatting issues.

This PR ignore one file (`content/zh/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions.md`) due to out-of-sync for ~500 lines.

## Related PRs
- #29847
- #32165

### CODE REVIEW

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md [hunk 1]
```diff
@@ -429,23 +429,23 @@ A cluster administrator can create a `RuntimeClass` object which is used to enca
...
+  nodeSelector:
+    kubernetes.io/os: 'windows'
+    kubernetes.io/arch: 'amd64'
+    node.kubernetes.io/windows-build: '10.0.17763'
+  tolerations:
+  - effect: NoSchedule
+    key: os
+    operator: Equal
+    value: "windows"
+```
```
Comment (tengqm) [inline content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md:448]: 这段还是额外缩进三个空格才对，因为它是前一个 item 的内容。

File: content/zh/docs/tasks/access-application-cluster/access-cluster.md [hunk 2]
```diff
@@ -49,11 +49,12 @@ kubectl config view
 ```
 
 <!--
-Many of the [examples](/docs/user-guide/kubectl-cheatsheet) provide an introduction to using
-`kubectl` and complete documentation is found in the [kubectl reference](/docs/reference/kubectl/).
+Many of the [examples](/docs/reference/kubectl/cheatsheet/) provide an introduction to using
+`kubectl`, and complete documentation is found in the
+[kubectl reference](/docs/reference/kubectl/).
 -->
 有许多 [例子](/zh/docs/reference/kubectl/cheatsheet/) 介绍了如何使用 kubectl，
```
Comment (tengqm) [inline content/zh/docs/tasks/access-application-cluster/access-cluster.md:56]:
```suggestion
有许多[例子](/zh/docs/reference/kubectl/cheatsheet/)介绍如何使用 kubectl，
```

File: content/zh/docs/tasks/access-application-cluster/access-cluster.md [hunk 3]
```diff
@@ -49,11 +49,12 @@ kubectl config view
...
-可以在 [kubectl 参考](/zh/docs/reference/kubectl/overview/) 中找到更完整的文档。
+可以在 [kubectl 参考](/zh/docs/reference/kubectl/) 中找到更完整的文档。
```
Comment (tengqm) [inline content/zh/docs/tasks/access-application-cluster/access-cluster.md:57]:
```suggestion
可以在 [kubectl 参考](/zh/docs/reference/kubectl/)中找到更完整的文档。
```

File: content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md (hunk 1 above)
Comment (Sea-n) [reply content/zh/docs/setup/production-environment/windows/user-guide-windows-containers.md:448]: Okay, maybe I should update en content instead.

File: content/zh/docs/tasks/access-application-cluster/access-cluster.md [hunk 4]
```diff
@@ -139,18 +146,18 @@ curl http://localhost:8080/api/
...
 First, create the Secret, requesting a token for the default ServiceAccount:
 
 -->
 
-### 不使用 kubectl proxy
+### 不使用 kubectl proxy {#without-kubectl-proxy}
 
-在 Kubernetes 1.3 或更高版本中，`kubectl config view` 不再显示 token。
 使用 `kubectl apply` 和 `kubectl describe secret ...` 及 grep 和剪切操作来为 default 服务帐户创建令牌，如下所示：
 `grep/cut` 方法实现：
```
Comment (tengqm) [inline content/zh/docs/tasks/access-application-cluster/access-cluster.md:158]: 删除 158 行

**Pull request merged**